src/Agent.Plugins/PipelineArtifact/PipelineArtifactServer.cs

```diff
@@ -79,90 +96,113 @@ public class PipelineArtifactServer
+                    else
+                    {
+                        context.Output(StringUtil.Loc("DownloadingMultiplePipelineArtifacts", pipelineArtifacts.Count()));
 
-                    var artifactNameAndManifestIds = pipelineArtifacts.ToDictionary(
-                        keySelector: (a) => a.Name, // keys should be unique, if not something is really wrong
-                        elementSelector: (a) => DedupIdentifier.Create(a.Resource.Data));
-                    // 2) download to the target path
-                    var options = DownloadPipelineArtifactOptions.CreateWithMultiManifestIds(
-                        artifactNameAndManifestIds,
-                        downloadParameters.TargetDirectory,
-                        proxyUri: null,
-                        minimatchPatterns: downloadParameters.MinimatchFilters);
-                    await buildDropManager.DownloadAsync(options, cancellationToken);
-                }
-            }
-            else if (downloadOptions == DownloadOptions.SingleDownload)
-            {
-                // 1) get manifest id from artifact data
-                BuildArtifact buildArtifact;
-                if (downloadParameters.ProjectRetrievalOptions == BuildArtifactRetrievalOptions.RetrieveByProjectId)
-                {
-                    buildArtifact = await buildHelper.GetArtifact(downloadParameters.ProjectId, downloadParameters.PipelineId, downloadParameters.ArtifactName, cancellationToken);
+                        var artifactNameAndManifestIds = pipelineArtifacts.ToDictionary(
+                            keySelector: (a) => a.Name, // keys should be unique, if not something is really wrong
+                            elementSelector: (a) => DedupIdentifier.Create(a.Resource.Data));
+                        // 2) download to the target path
+                        var options = DownloadPipelineArtifactOptions.CreateWithMultiManifestIds(
+                            artifactNameAndManifestIds,
+                            downloadParameters.TargetDirectory,
+                            proxyUri: null,
+                            minimatchPatterns: downloadParameters.MinimatchFilters);
+
+                        PipelineArtifactActionRecord downloadRecord = clientTelemetry.CreateRecord<PipelineArtifactActionRecord>((level, uri, type) =>
+                            new PipelineArtifactActionRecord(level, uri, type, nameof(DownloadAsync), context));
+                        await clientTelemetry.MeasureActionAsync(
+                            record: downloadRecord,
+                            actionAsync: async () =>
+                            {
+                                await dedupManifestClient.DownloadAsync(options, cancellationToken);
+                            });
+                        // Send results to CustomerIntelligence
+                        context.PublishTelemetry(area: "AzurePipelinesAgent", feature: "PipelineArtifact", record: downloadRecord);
```

Comment:
>"AzurePipelinesAgent" [](start = 55, length = 21)

this is declared as a constant in PipelineCacheServer. Can this be shared in a constants file?